### PR TITLE
refactor: replace glog in main.go

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

- remove google/glog from main.go

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: cmd/nginx-ingress/main.go [hunk 1]
```diff
@@ -705,34 +719,36 @@ func createManagerAndControllerCollectors(constLabels map[string]string) (collec
 
 		err = mc.Register(registry)
 		if err != nil {
-			glog.Errorf("Error registering Manager Prometheus metrics: %v", err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error registering Manager Prometheus metrics: %v", err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:722]:
```suggestion
			l.Error(fmt.Sprintf("Error registering Manager Prometheus metrics: %v", err))
```

File: cmd/nginx-ingress/main.go [hunk 2]
```diff
@@ -705,34 +719,36 @@ func createManagerAndControllerCollectors(constLabels map[string]string) (collec
...
 		}
 
 		err = cc.Register(registry)
 		if err != nil {
-			glog.Errorf("Error registering Controller Prometheus metrics: %v", err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error registering Controller Prometheus metrics: %v", err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:727]:
```suggestion
			l.Error(fmt.Sprintf("Error registering Controller Prometheus metrics: %v", err))
```

File: cmd/nginx-ingress/main.go [hunk 3]
```diff
@@ -705,34 +719,36 @@ func createManagerAndControllerCollectors(constLabels map[string]string) (collec
...
 		}
 
 		err = processCollector.Register(registry)
 		if err != nil {
-			glog.Errorf("Error registering NginxProcess Prometheus metrics: %v", err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error registering NginxProcess Prometheus metrics: %v", err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:732]:
```suggestion
			l.Error(fmt.Sprintf("Error registering NginxProcess Prometheus metrics: %v", err))
```

File: cmd/nginx-ingress/main.go [hunk 4]
```diff
@@ -705,34 +719,36 @@ func createManagerAndControllerCollectors(constLabels map[string]string) (collec
...
 		}
 
 		err = workQueueCollector.Register(registry)
 		if err != nil {
-			glog.Errorf("Error registering WorkQueue Prometheus metrics: %v", err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error registering WorkQueue Prometheus metrics: %v", err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:737]:
```suggestion
			l.Error(fmt.Sprintf("Error registering WorkQueue Prometheus metrics: %v", err))
```

File: cmd/nginx-ingress/main.go [hunk 5]
```diff
@@ -774,7 +790,7 @@ func createPlusAndLatencyCollectors(
 		if *enableLatencyMetrics {
 			lc = collectors.NewLatencyMetricsCollector(constLabels, upstreamServerVariableLabels, upstreamServerPeerVariableLabelNames)
 			if err := lc.Register(registry); err != nil {
-				glog.Errorf("Error registering Latency Prometheus metrics: %v", err)
+				l.Log(ctx, levels.LevelError, fmt.Sprintf("Error registering Latency Prometheus metrics: %v", err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:793]:
```suggestion
				l.Error(fmt.Sprintf("Error registering Latency Prometheus metrics: %v", err))
```

File: cmd/nginx-ingress/main.go [hunk 6]
```diff
@@ -859,7 +879,7 @@ func updateSelfWithVersionInfo(kubeClient *kubernetes.Clientset, version, appPro
 		}
 		pod, err := kubeClient.CoreV1().Pods(os.Getenv("POD_NAMESPACE")).Get(context.TODO(), os.Getenv("POD_NAME"), meta_v1.GetOptions{})
 		if err != nil {
-			glog.Errorf("Error getting pod on attempt %d of %d: %v", i+1, maxRetries, err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error getting pod on attempt %d of %d: %v", i+1, maxRetries, err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:882]:
```suggestion
			l.Error(fmt.Sprintf("Error getting pod on attempt %d of %d: %v", i+1, maxRetries, err))
```

File: cmd/nginx-ingress/main.go [hunk 7]
```diff
@@ -882,16 +902,16 @@ func updateSelfWithVersionInfo(kubeClient *kubernetes.Clientset, version, appPro
 
 		_, err = kubeClient.CoreV1().Pods(newPod.ObjectMeta.Namespace).Update(context.TODO(), newPod, meta_v1.UpdateOptions{})
 		if err != nil {
-			glog.Errorf("Error updating pod with labels on attempt %d of %d: %v", i+1, maxRetries, err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("Error updating pod with labels on attempt %d of %d: %v", i+1, maxRetries, err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:905]:
```suggestion
			l.Error(fmt.Sprintf("Error updating pod with labels on attempt %d of %d: %v", i+1, maxRetries, err))
```

File: cmd/nginx-ingress/main.go [hunk 8]
```diff
@@ -882,16 +902,16 @@ func updateSelfWithVersionInfo(kubeClient *kubernetes.Clientset, version, appPro
...
 		}
 
-		glog.Infof("Pod label updated: %s", pod.ObjectMeta.Name)
+		l.Info(fmt.Sprintf("Pod label updated: %s", pod.ObjectMeta.Name))
 		podUpdated = true
 	}
 
 	if !podUpdated {
-		glog.Errorf("Failed to update pod labels after %d attempts", maxRetries)
+		l.Log(ctx, levels.LevelError, fmt.Sprintf("Failed to update pod labels after %d attempts", maxRetries))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:914]:
```suggestion
		l.Error(fmt.Sprintf("Failed to update pod labels after %d attempts", maxRetries))
```

**Review by pdabelf5 — COMMENTED** [review]
🚀

**Review by jjngx — APPROVED** [review]

File: cmd/nginx-ingress/main.go [hunk 9]
```diff
@@ -262,21 +261,23 @@ func main() {
 			port := fmt.Sprintf(":%v", *readyStatusPort)
 			s := http.NewServeMux()
 			s.HandleFunc("/nginx-ready", ready(lbc))
-			glog.Fatal(http.ListenAndServe(port, s))
+			l.Log(ctx, levels.LevelFatal, fmt.Sprint(http.ListenAndServe(port, s))) // nolint:gosec
+			os.Exit(0)
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:265]:
```suggestion
			os.Exit(1)
```
Exit 0 is a success exit code

File: cmd/nginx-ingress/main.go [hunk 10]
```diff
@@ -288,125 +289,144 @@ func mustCreateConfigAndKubeClient() (*rest.Config, *kubernetes.Clientset) {
...
-func checkNamespaces(kubeClient kubernetes.Interface) {
+func checkNamespaces(ctx context.Context, kubeClient kubernetes.Interface) {
+	l := nic_logger.LoggerFromContext(ctx)
 	if *watchNamespaceLabel != "" {
 		// bootstrap the watched namespace list
 		var newWatchNamespaces []string
 		nsList, err := kubeClient.CoreV1().Namespaces().List(context.TODO(), meta_v1.ListOptions{LabelSelector: *watchNamespaceLabel})
 		if err != nil {
-			glog.Errorf("error when getting Namespaces with the label selector %v: %v", watchNamespaceLabel, err)
+			l.Log(ctx, levels.LevelError, fmt.Sprintf("error when getting Namespaces with the label selector %v: %v", watchNamespaceLabel, err))
```
Comment (pdabelf5) [inline cmd/nginx-ingress/main.go:357]:
```suggestion
			l.Error(fmt.Sprintf("error when getting Namespaces with the label selector %v: %v", watchNamespaceLabel, err))
```

**Review by pdabelf5 — APPROVED** [review]

**Pull request merged**